hacking/release-procedure.rst

```diff
@@ -0,0 +1,350 @@
+    * Ping ``@dharmabumstead`` the release announcement link
+
+* Update the changelog
+
+    * People should have been updating the changelog as they added major features.  Ping them to
+      make sure they've done that
+    * Cut and paste from the previous stable-X.Y changelog into devel so that devel has a record of
+      what was in the X.Y minor releases.
+
+
+To make a final major release
+-----------------------------
+
+* In the ``devel`` branch edit :file:`packaging/release/vars/versions.yml` -- Update the release
+  date
+* In both ``devel`` and ``stable-2.4`` branch edit release status in
+  :file:`docs/docsite/rst/release_and_maintenance.rst`
+* Steps 4-6 of the :ref:`beta1 <first_release>` section
+* Additional final release Sanity checks
+
+   * :file:`RELEASES.txt`
+   * :file:`packaging/rpm/ansible.spec`
+   * :file:`packaging/debian/changelog`
+
+* Steps 7-9 of the :ref:`beta1 <first_release>` section
+* Build and upload the tarballs/rpm/deb packages via jenkins:
+
+   * http://jenkins.testing.ansible.com/view/Ansible/job/Release_Ansible/
+   * GIT_BRANCH=origin/tags/v2.4.0.0-1
+   * CONFIRM=<check it>
+   * DEB_PPA=<leave as-is>
+   * NOTE: the jenkins job does not generate sha256sum files
+
+* Steps 12-13 of the :ref:`beta1 <first release>` section
+* Other things to check:
+
+   * New release on https://pypi.python.org/pypi/ansible
+   * New release on https://releases.ansible.com/ansible
+   * New release on the PPA https://launchpad.net/~ansible/+archive/ubuntu/ansible
+   * Old release on the PPA https://launchpad.net/~ansible/+archive/ubuntu/ansible-2.3
+   * New release in the rpms directory: http://releases.ansible.com/ansible/rpm/release/
+
+* Send email to ``ansible-announce`` and ``ansible-project`` googlegroup
+
+   * This should have been worked on with the rest of the team prior to the final rc.
+   * Get the email link from the `googlegroup archive
+     <https://groups.google.com/forum/#!forum/ansible-announce>`_
+
+* Steps 16 of the :ref:`beta1 <first_release>` section
+
+   * NOTE: We do not tweet the final releases.  Marketing handles that entirely
+
+* Send a message to #ship_it on slack to alert tower team that the final release is out
+* Email ansible-tower@redhat.com to alert tower team of final release
+* Run the Jenkins DEB job to upload the latest 2.3 version to this PPA too
+
+   * http://jenkins.testing.ansible.com/job/Build_Ansible_DEB/
+   * GIT_BRANCH=v2.3.2.0-1
+   * OFFICIAL=yes
+   * DEB_DIST=<leave as is>
+   * DEB_PPA=ppa:ansible/ansible-2.3
+
+* Merge the ``temp-staging-post-2.4.0`` branch back into the ``stable-2.4`` branch and then remove
+  the branch from the repo
+
+   * Relax permissions on the stable-2.4 branch so that anyone can commit again
+
+
+To make a dot release
+---------------------
+
+* Step 4-16 of the :ref:`beta1 <first_release>` process
+
+
+Building docs
+-------------
+
+.. Should this be moved to its own page?  And then link to it at the points where docs should be
```

Comment:
I think it would be good to have the docs for `stable-X.Y` and hopefully `devel` complete before sending the email.

+1 to killing other GitHub triggered jobs

Reply:
Once docs builds are a reasonable time frame I could be okay with that.  As it is, it's not practical to ask the RM to wait until docs are built.  Docs building is also isolated from the release process.  Perhaps we should give someone the task of building the docs when they hear certain things from the RM.  Like: stable-2.4 has been branched/Beta1 has been released.  RC1 has been released.  (RC1 is a good trigger if no further docs changes are merged after RC1.  Perhaps the RM should also be responsible for alerting the Docs Master when any PRs touching docs are released merged after RC1)?